chutney/action-impl/src/main/java/com/chutneytesting/action/kafka/KafkaBasicPublishAction.java

```diff
@@ -40,18 +40,30 @@ public class KafkaBasicPublishAction implements Action {
     private final Map<String, String> headers;
     private final String payload;
     private final Map<String, String> properties;
+    private final String key;
     private final Logger logger;
 
+    public KafkaBasicPublishAction(Target target,
+                                   @Input("topic") String topic,
```

Comment:
In which case do still need this? if key is optional?

Reply:
Right now key is optional, you prefer to make it required and to remove topic ?

Reply:
key should be Optional